course7/homeworks/patrickkostjens/ls/ls.cc

```diff
@@ -1,13 +1,62 @@
 #include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
 #include <iostream>
 
+namespace fs = boost::filesystem;
+namespace po = boost::program_options;
+
+struct node_result
+{
+	std::string output;
+	size_t size;
+};
+
+node_result* print_contents(const fs::path& path, bool extensive, bool tree, int level = 0)
+{
+	node_result* result = new node_result();
```

Comment:
Don't! This is C#-ism. Just create the object on the stack and return it as if it was a normal value.
Please mote: you never delete the object, leaking resources!